cmd/ci-reporter/cmd/root.go

```diff
@@ -178,19 +191,50 @@ func (r *CIReporters) CollectReportDataFromReporters(cfg *Config) (*CIReportData
 }
 
 // PrintReporterData used to print report data
+// 1. Get a output stream to write the data to
+// 2. Write data to stream
+// 	2.1. Write data in JSON format if set so
+// 	2.2. Write data in table format
 func PrintReporterData(cfg *Config, reports *CIReportDataFields) error {
+	// Get a stream to write the data to (file stream / standard out stream)
+	var out *os.File
+	if cfg.Filepath != "" {
+		// open output file
+		fileOut, err := os.OpenFile(cfg.Filepath, os.O_WRONLY|os.O_CREATE, 0o666)
+		if err != nil {
+			return errors.Wrapf(err, "could not open or create a file at %s to write the ci signal report to", cfg.Filepath)
+		}
+		out = fileOut
+	} else {
+		out = os.Stdout
+	}
+	defer func() {
+		if err := out.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// Write data to stream
 	if cfg.JSONOutput {
 		// print report in json format
 		d, err := reports.Marshal()
 		if err != nil {
 			return nil
 		}
-		fmt.Print(string(d))
+		_, err = out.WriteString(string(d))
+		if err != nil {
+			return errors.Wrap(err, "could not write to output stream")
+		}
```

Comment:
Keep this if block but then return early and remove the else block bellow to simplify

Reply:
Okay, sounds good! I also fixed line 222 to return the error if marshalling failed